src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -1476,29 +1476,133 @@ def _configure_security_policy_storage_params(arg_ctx):
 
     with self.argument_context('sql midb short-term-retention-policy show') as c:
-        c.argument('deleted_time',
-                   options_list=['--deleted-time'],
-                   help='If specified, shows retention days for a deleted database, instead of an existing database.'
-                   'Must match the deleted time of a deleted database on the source Managed Instance.')
+        c.argument(
+            'deleted_time',
+            options_list=['--deleted-time'],
+            help='If specified, shows retention days for a deleted database, instead of an existing database.'
+            'Must match the deleted time of a deleted database on the source Managed Instance.')
+
+    with self.argument_context('sql midb long-term-retention-policy set') as c:
+        create_args_for_complex_type(
+            c, 'parameters', ManagedDatabase, [
+                'weekly_retention',
+                'monthly_retention',
+                'yearly_retention',
+                'week_of_year'
+            ])
+
+        c.argument('weekly_retention',
+                   help='The Weekly Retention.'
+                   'If just a number is passed instead of an ISO 8601 string, days will be assumed as the units.'
+                   'There is a minimum of 7 days and a maximum of 10 years.')
+
+        c.argument('monthly_retention',
+                   help='The Monthly Retention.'
+                   'If just a number is passed instead of an ISO 8601 string, days will be assumed as the units.'
+                   'There is a minimum of 7 days and a maximum of 10 years.')
+
+        c.argument('yearly_retention',
+                   help='The Yearly Retention.'
+                   'If just a number is passed instead of an ISO 8601 string, days will be assumed as the units.'
+                   'There is a minimum of 7 days and a maximum of 10 years.')
+
+        c.argument('week_of_year',
+                   help='The Week of Year, 1 to 52, to save for the Yearly Retention.')
+
+    with self.argument_context('sql midb long-term-retention-backup') as c:
+        c.argument('location_name',
+                   required=True,
+                   arg_type=get_location_type(self.cli_ctx))
+
+        c.argument('resource_group_name',
+                   required=False,
+                   help='If specified, the resource group that the managed instance/database resource belongs to.')
+
+    with self.argument_context('sql midb long-term-retention-backup show') as c:
+        c.argument('backup_name',
+                   options_list=['--backup-name', '-b'],
+                   required=True,
+                   help='The long term retention backup name.')
+
+    with self.argument_context('sql midb long-term-retention-backup list-by-database') as c:
+        c.argument('database_state',
+                   required=False,
+                   help='\'All\', \'Live\', or \'Deleted\'')
+
+        c.argument('only_latest_per_database',
```

Comment:
`--only-latest-per-database` is too long as parameter. Can you give a short name for this parameter?

Reply:
noted

Reply:
So for this argument, we will assume it to be true when provided, right? If yes, can we consider using `action='store_true'` for this argument to make sure it will be true when specified?

Reply:
We will assume false unless provided.